service-spec/service-spec-commons/src/main/java/io/hyscale/servicespec/commons/model/service/ReplicasDeserializer.java

```diff
@@ -44,22 +44,22 @@
 
     @Override
     public Object deserialize(JsonParser jsonParser, DeserializationContext ctxt) throws IOException, JsonProcessingException {
-        JsonNode specNode = jsonParser.readValueAsTree();
-        if (specNode == null) {
-            return null;
-        }
-        Replicas replicas = new Replicas();
-        JsonNode rootNode = specNode.get(HyscaleSpecFields.replicas);
-        if (rootNode == null) {
+        JsonNode replicasNode = jsonParser.readValueAsTree();
+        // specNode refers to replicas
+        if (replicasNode == null) {
             return defaultReplicas();
         }
+        Replicas replicas = new Replicas();
         try {
-            if (rootNode.get(HyscaleSpecFields.min) != null) {
-                replicas.setMin(Integer.valueOf(rootNode.get(HyscaleSpecFields.min).toString()));
-                replicas.setMax(Integer.valueOf(rootNode.get(HyscaleSpecFields.max).toString()));
-                replicas.setCpuThreshold(rootNode.get(HyscaleSpecFields.cpuThreshold).toString());
+            if (replicasNode.get(HyscaleSpecFields.min) != null) {
+                replicas.setMin(Integer.valueOf(replicasNode.get(HyscaleSpecFields.min).toString()));
+                replicas.setMax(Integer.valueOf(replicasNode.get(HyscaleSpecFields.max).toString()));
+                JsonNode cpuThreshold = replicasNode.get(HyscaleSpecFields.cpuThreshold);
+                if (cpuThreshold != null && !cpuThreshold.isNull()) {
+                    replicas.setCpuThreshold(cpuThreshold.textValue());
```

Comment:
Null checks for max value.

Reply:
In case of null, Integer.valueOf will throw NumberFormatException which we catch and return default replicas value. Should I add a explicit check?